spring-data-cassandra/src/main/java/org/springframework/data/cassandra/repository/support/CassandraRepositoryFragmentsContributor.java

```diff
@@ -0,0 +1,98 @@
+import org.springframework.data.repository.core.support.RepositoryFragmentsContributor;
+import org.springframework.util.Assert;
+
+/**
+ * Cassandra-specific {@link RepositoryFragmentsContributor} contributing fragments based on the repository.
+ * <p>
+ * Implementations must define a no-args constructor.
+ *
+ * @author Chris Bono
+ * @since 5.0
+ */
+public interface CassandraRepositoryFragmentsContributor extends RepositoryFragmentsContributor {
+
+	CassandraRepositoryFragmentsContributor DEFAULT = DefaultCassandraRepositoryFragmentsContributor.INSTANCE;
+
+	/**
+	 * Returns a composed {@code CassandraRepositoryFragmentsContributor} that first applies this contributor to its inputs,
+	 * and then applies the {@code after} contributor concatenating effectively both results. If evaluation of either
+	 * contributors throws an exception, it is relayed to the caller of the composed contributor.
+	 *
+	 * @param after the contributor to apply after this contributor is applied.
+	 * @return a composed contributor that first applies this contributor and then applies the {@code after} contributor.
+	 */
+	default CassandraRepositoryFragmentsContributor andThen(CassandraRepositoryFragmentsContributor after) {
+
+		Assert.notNull(after, "CassandraRepositoryFragmentsContributor must not be null");
+
+		return new CassandraRepositoryFragmentsContributor() {
+
+			@Override
+			public RepositoryFragments contribute(RepositoryMetadata metadata,
+					CassandraEntityInformation<?, ?> entityInformation, CassandraOperations operations) {
+				return CassandraRepositoryFragmentsContributor.this.contribute(metadata, entityInformation, operations)
+						.append(after.contribute(metadata, entityInformation, operations));
+			}
+
+			@Override
+			public RepositoryFragments describe(RepositoryMetadata metadata) {
+				return CassandraRepositoryFragmentsContributor.this.describe(metadata).append(after.describe(metadata));
+			}
+		};
+	}
+
+	/**
+	 * Creates {@link RepositoryFragments} based on {@link RepositoryMetadata} to add
+	 * Cassandra-specific extensions.
+	 *
+	 * @param metadata repository metadata.
+	 * @param entityInformation must not be {@literal null}.
+	 * @param operations must not be {@literal null}.
+	 * @return {@link RepositoryFragments} to be added to the repository.
+	 */
+	RepositoryFragments contribute(RepositoryMetadata metadata,
+			CassandraEntityInformation<?, ?> entityInformation, CassandraOperations operations);
+
+	/**
+	 * Implementation of {@link CassandraRepositoryFragmentsContributor} that contributes empty fragments by default.
+	 *
+	 * @author Chris Bono
+	 * @since 5.0
+	 */
+	enum DefaultCassandraRepositoryFragmentsContributor implements CassandraRepositoryFragmentsContributor {
+
+		INSTANCE;
+
+		@Override
+		public RepositoryFragments contribute(RepositoryMetadata metadata,
+				CassandraEntityInformation<?, ?> entityInformation, CassandraOperations operations) {
+			return RepositoryFragments.empty();
```

Comment:
Eventhough the fragment are empty by default for Cassandra, I chose to create a default impl that simple returned empty() to serve as a non-null fragmentsContributor default in the factory/factoryBeans as well as a placeholder in case we add default fragments (e.g. the MongoDB queryDsl fragments).

Reply:
Makes sense to have the same extension points across our Spring Data modules. Having the enum inside the interface makes it public. I suggest having a package-protected visibility for the default implementation.